src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -132,6 +132,21 @@ public final class InlineConfigParser {
     private static final Pattern VIOLATION_SOME_LINES_BELOW_PATTERN = Pattern
             .compile(".*//\\s*violation (\\d+) lines below\\s*(?:['\"](.*)['\"])?$");
 
+    /**
+     * <p>
+     * Multiple violations for above line. Messages are X lines below.
+     * {@code
+     *   // X violations above:
+     *   //                    'violation message1'
+     *   //                    'violation messageX'
+     * }
+     *
+     * Messages are matched by {@link InlineConfigParser#VIOLATION_MESSAGE_PATTERN}
+     * </p>
+     */
+    private static final Pattern VIOLATIONS_ABOVE_PATTERN_WITH_MESSAGES = Pattern
+            .compile(".*//\\s*(\\d+) violations above:\\s*(?:['\"](.*)['\"])?$");
```

Comment:
What is the purpose of `(?:['\"](.*)['\"])?` ? I don't see it being used anywhere and it seems to be all over the place in this class.

Reply:
Good point, we do not use it, we need to remove it.

Reply:
In all other places, it is to get a message from line and match it to what Check is generating

Reply:
fixed.

Reply:
Not really. Now that you removed the rest, it now raises why do we need `\\s*` on the end? We don't allow trailing spaces, and `*` says it is completely optional. So it should be removed as well.

Reply:
fixed.